src/linter.rs

```diff
@@ -70,6 +70,12 @@ pub(crate) fn check_path(
     // Aggregate all checks.
     let mut checks: Vec<Check> = vec![];
 
+    if stylist.line_ending() == &LineEnding::Cr {
+        return Err(anyhow!(
+            "Detected old MacOS style line-endings that are unsupported."
+        ));
+    }
```

Comment:
Is there any way to do this lazily? Like, detect when it would become a problem, and fail in that case? (Not asking you to make that change.)

Reply:
Actually, I guess it's fine, this should be quite cheap...